File: Dev/unitTest_Engine_cpp_gtest/Input/MouseWithManyButtons.cpp
#include "../EngineTest.h"
#include <Altseed.h>
#include <gtest/gtest.h>

using namespace std;
using namespace asd;

class Input_MouseWithManyButtons : public EngineTest
{
	vector<shared_ptr<TextObject2D>> textObjects;

public:
	Input_MouseWithManyButtons(bool isOpenGLMode) : EngineTest(u"MouseWithManyButtons", isOpenGLMode, 300) {}

protected:
	void OnStart()
	{
		auto scene = make_shared<Scene>();
		auto layer = make_shared<Layer2D>();

		for (int i = 0; i < 8; ++i)
		{
			auto text = make_shared<TextObject2D>();
			text->SetFont(
				Engine::GetGraphics()->CreateDynamicFont(asd::ToAString("").c_str(), 32, Color(255, 0, 0, 255), 0, Color(0, 0, 0, 0)));
			text->SetPosition(Vector2DF(40, 40 + 50 * i));
			textObjects.push_back(text);
			layer->AddObject(text);
		}

		scene->AddLayer(layer);
		asd::Engine::ChangeScene(scene);
	}

	void OnUpdating()
	{
		for (int i = 0; i < 8; ++i)
		{
			auto inputState = Engine::GetMouse()->GetButtonInputState((MouseButtons)i);

			if (inputState == ButtonState::Push)
			{
				textObjects[i]->SetText((ToAString("ボタン") + ToAString(to_string(i).c_str()) + ToAString("を押しました。")).c_str());
			}
			else if (inputState == ButtonState::Hold)
			{
				textObjects[i]->SetText(
					(ToAString("ボタン") + ToAString(to_string(i).c_str()) + ToAString("を押し続けています。")).c_str());
			}
			else if (inputState == ButtonState::Free)
			{
				textObjects[i]->SetText(
					(ToAString("ボタン") + ToAString(to_string(i).c_str()) + ToAString("を離し続けています。")).c_str());
			}
			else if (inputState == ButtonState::Release)
			{
				textObjects[i]->SetText((ToAString("ボタン") + ToAString(to_string(i).c_str()) + ToAString("を離しました。")).c_str());
			}
		}
	}
};

ENGINE_TEST(Input, MouseWithManyButtons)

File: Dev/unitTest_Engine_cs/Input/MouseWithManyButtons.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using asd;

namespace unitTest_Engine_cs.Input
{
    class MouseWithManyButtons : EngineTest
    {
        private List<TextObject2D> textObjects;

        public MouseWithManyButtons()
            : base(300)
        {
        }

        protected override void OnStart()
        {
            var scene = new Scene();
            var layer = new Layer2D();

            textObjects = new List<TextObject2D>();
            for (int i = 0; i < 8; ++i)
            {
                var text = new TextObject2D();
                text.Font = Engine.Graphics.CreateDynamicFont("", 32, new Color(255, 0, 0, 255), 0, new Color(0, 0, 0, 0));
                text.Position = new Vector2DF(40, 40 + 50 * i);
                textObjects.Add(text);
                layer.AddObject(text);
            }

            scene.AddLayer(layer);
            asd.Engine.ChangeScene(scene);
        }

        protected override void OnUpdating()
        {
            for (int i = 0; i < 8; ++i)
            {
                var inputState = Engine.Mouse.GetButtonInputState((MouseButtons)i);

                if (inputState == ButtonState.Push)
                {
                    textObjects[i].Text = "ボタン" + i + "を押しました。";
                }
                else if (inputState == ButtonState.Hold)
                {
                    textObjects[i].Text = "ボタン" + i + "を押し続けています。";
                }
                else if (inputState == ButtonState.Free)
                {
                    textObjects[i].Text = "ボタン" + i + "を離し続けています。";
                }
                else if (inputState == ButtonState.Release)
                {
                    textObjects[i].Text = "ボタン" + i + "を離しました。";
                }
            }
        }
    }
}
